controllers/component_dependency_update_controller.go

```diff
@@ -618,13 +618,23 @@ func generateRenovateConfigForNudge(slug string, repositories []renovateReposito
 // See https://issues.redhat.com/browse/KFLUXBUGS-1233
 // This will map repsitories of the form 'quay.io/redhat-prod/foo----bar' to 'registry.redhat.io/foo/bar'
 func mapToRegistryRedhatIo(repo string) string {
-
-	regex, err := regexp.Compile(`^quay.io/redhat-prod/(.*)----(.*)$`)
+	prodRegex, err := regexp.Compile(`^quay.io/redhat-prod/(.*)----(.*)$`)
 	if err != nil {
 		return ""
 	}
-	results := regex.FindStringSubmatch(repo)
+
+	results := prodRegex.FindStringSubmatch(repo)
 	if results == nil {
```

Comment:
nit: I think it would be better to return early with 
```golang
if results != nil {
    return "registry.redhat.io/" + results[1] + "/" + results[2]
}
```
 so the stage regex logic would not be nested in the if